torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1056,6 +1058,51 @@ void ProcessGroupNCCL::waitForDumpOrTimeout(
   std::this_thread::sleep_until(wakeUpTime);
 }
 
+void ProcessGroupNCCL::waitForFutureOrTimeout(
```

Comment:
nice helper. maybe in a followup PR you can replace my other wait functions with this? i think its better to keep that separate though.

Reply:
Yes, this was on my list ^_^.